crypto/mem_sec.c

```diff
@@ -260,11 +260,19 @@ int CRYPTO_secure_allocated(const void *ptr)
 
 size_t CRYPTO_secure_used(void)
 {
+    size_t ret = 0;
+
 #ifndef OPENSSL_NO_SECURE_MEMORY
-    return secure_mem_used;
+    if (!CRYPTO_THREAD_read_lock(sec_malloc_lock))
+        return 0;
+
+    ret = secure_mem_used;
+
+    CRYPTO_THREAD_unlock(sec_malloc_lock);
 #else
-    return 0;
+    ret = 0;
```

Comment:
do you need the `#else ... #endif` block?

Reply:
Its there, that line just wasn't changed

Reply:
the way you have refactored this, I think you don't need the `#else ... #endif` block at all.

lines 273-275 in your updated version could just be replaced with `#endif`.

Reply:
ah, I see what you're saying, yeah, I can remove that

Reply:
Please do

Reply:
fixed